UnicornEngine/video/source/vulkan/Renderer.cpp

```diff
@@ -414,10 +452,13 @@ void Renderer::DeleteVkMesh(VkMesh* pVkMesh)
 
 void Renderer::FreeSurface()
 {
-    if(m_vkWindowSurface && m_vkLogicalDevice)
+    if(m_vkLogicalDevice)
     {
-        Context::Instance().GetVkInstance().destroySurfaceKHR(m_vkWindowSurface);
-        m_vkWindowSurface = nullptr;
+        if(m_vkWindowSurface)
+        {
+            Context::Instance().GetVkInstance().destroySurfaceKHR(m_vkWindowSurface);
```

Comment:
If `Context::Instance()` can't be changed during runtime, please store it in some class variable during initialization